### DISCUSSION TITLE
A delay using the sidekiq scheduler?

### DISCUSSION THREAD

**Discussion by ericgross** [RailsEventStore/rails_event_store#1713] [Q&A] [answered]
I am able to publish asynchronous jobs which get run the sidekiq integration. I would like to publish some commands that are sent to sidekiq for execution after a certain delay. Is it possible to specify a delay when using the sidekiq scheduler in order to have the job not run for a second period of time and then execute?

**Comment by pjurewicz** [comment, marked as answer]
Hello @ericgross,
assuming that you are using RES 2.12+, the answer is YES.

Sidekiq scheduler now accepts `Sidekiq::Job::Setter` instance as a valid handler. So that you can set up your subscription like this:
```event_store.subscribe(NotifyJob.set(wait: 5.seconds), to: [StatusChanged])```
